File: Data Structures & Algorithms/balanced-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    bool isB = true;
    bool isBalanced(TreeNode* root) {
        depth(root);
        return isB;
    }

    int depth(TreeNode* root)
    {
        if (root == nullptr)
            return 0;
        
        auto leftDepth = depth(root->left);
        auto rightDepth = depth(root->right);
        
        if (std::abs(leftDepth - rightDepth) > 1)
            isB = false;
        return 1 + max(leftDepth, rightDepth);
    };

};

File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if root is None:
            return 0

        node = root
        stack = [(root, 1)]
        maxSize = 0

        while stack:
            node, depth = stack.pop()

            maxSize = max(maxSize, depth)

            if node.left is not None:
                stack.append((node.left, depth+1))
            if node.right is not None:
                stack.append((node.right, depth+1))
        return maxSize
